src/parser/scanner.lex

```diff
@@ -249,20 +249,64 @@ IP_OCTET                    ([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])
                                 return TokenType::IPV4;
                             }
 0[Xx]{HEX}+                 {
+                                if (yyleng > 18) {
+                                    auto i = 2;
+                                    while (i < yyleng && yytext[i] == '0') {
+                                        i++;
+                                    }
+                                    if (yyleng - i > 16) {
+                                        yyterminate();
+                                    }
```

Comment:
The maximum value of int64 is 9,223,372,036,854,775,807,
Hexadecimal is 0X7FFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF
when value is in [ 0X8000 0000 0000 0000 0000 0000 0000 0000, 0XFFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF] ， the value is still overflow.

So, here add esle statement

Reply:
Hex is a way to specify integers as raw bytes, there is no such thing overflow in terms of literals.

Reply:
if （yyleng - i == 16 && yytext[i] > 7 ),   Does the hexadecimal  value not overflow?
the value in decimal is great than 9,223,372,036,854,775,807.

Reply:
Yeah， you are right,   [-9,223,372,036,854,775,808     9,223,372,036,854,775,807] is [0, 0XFFFF FFFF FFFF FFFF]
👍

Reply:
you can replace  magic number "18" with  std::numeric_limits<int64_t>::digits10 .

Reply:
@monadbobo But here we are dealing with hexdecimal, instead of base 10.